plugins/scaffolder/src/components/ScaffolderPage/ScaffolderPage.tsx

```diff
@@ -93,33 +145,59 @@ export const ScaffolderPage = () => {
             documentation, ...).
           </SupportButton>
         </ContentHeader>
-        {!templates && isValidating && <Progress />}
-        {templates && !templates.length && (
-          <Typography variant="body2">
-            Shoot! Looks like you don't have any templates. Check out the
-            documentation{' '}
-            <Link href="https://backstage.io/docs/features/software-templates/adding-templates">
-              here!
-            </Link>
-          </Typography>
-        )}
-        {error && (
-          <WarningPanel>
-            Oops! Something went wrong loading the templates: {error.message}
-          </WarningPanel>
-        )}
-        <Grid container>
-          {templates &&
-            templates?.length > 0 &&
-            templates.map(template => {
-              return (
-                <Grid key={template.metadata.uid} item xs={12} sm={6} md={3}>
-                  <TemplateCard {...getTemplateCardProps(template)} />
-                </Grid>
-              );
-            })}
-        </Grid>
+
+        <div className={styles.contentWrapper}>
+          <div>
+            <SearchToolbar search={search} setSearch={setSearch} />
+            <ScaffolderFilter
+              buttonGroups={filterGroups}
+              initiallySelected="all"
+            />
+            <ResultsFilter availableCategories={availableCategories} />
+          </div>
+          <div>
+            {!matchingEntities && loading && <Progress />}
+            {matchingEntities && !matchingEntities.length && (
+              <Typography variant="body2">
+                Shoot! Looks like you don't have any templates. Check out the
+                documentation{' '}
+                <Link href="https://backstage.io/docs/features/software-templates/adding-templates">
+                  here!
+                </Link>
+              </Typography>
+            )}
+            {error && (
+              <WarningPanel>
+                Oops! Something went wrong loading the templates:{' '}
+                {error.message}
+              </WarningPanel>
```

Comment:
👍 for this!  I think ideally you provide the "Oops! Something went wrong loading the templates." as a `title={}` param, and leave the error.message for the child element inside the component. That way, the folded header says "Warning: Oops! Something went wrong loading the templates." and the user can click it to unfold to see the error?

Reply:
Sure! Makes more sense. Updating it

Reply:
Fixed in 452602094f23e5bb0eb150e649c9661066d2f75e